python/ray/serve/_private/logging_utils.py

```diff
@@ -74,12 +75,20 @@ def format(self, record: logging.LogRecord) -> str:
             ]
         if SERVE_LOG_ROUTE in record.__dict__:
             record_format[SERVE_LOG_ROUTE] = SERVE_LOG_RECORD_FORMAT[SERVE_LOG_ROUTE]
+
         if SERVE_LOG_APPLICATION in record.__dict__:
             record_format[SERVE_LOG_APPLICATION] = SERVE_LOG_RECORD_FORMAT[
                 SERVE_LOG_APPLICATION
             ]
 
-        record_format[SERVE_LOG_MESSAGE] = SERVE_LOG_RECORD_FORMAT[SERVE_LOG_MESSAGE]
+        if SERVE_LOG_EXTRA_FIELDS in record.__dict__:
+            for k, v in record.__dict__[SERVE_LOG_EXTRA_FIELDS].items():
```

Comment:
Are there mechanisms to ensure this is always a dictionary?

Reply:
add a check and raise error.